### ISSUE TITLE
Error in checkForHole function?

### ISSUE THREAD

**Issue by caewok** [theripper93/Levels#30]
This is the current `checkForHole` function:
```js
    //Check if a floor is hollowed by a hole
    function checkForHole(intersectionPT, zz) {
      for (let hole of _levels.levelsHoles) {
        const hbottom = hole.range[0];
        const htop = hole.range[1];
        if (zz > htop || zz < hbottom) continue;
        if (hole.poly.contains(intersectionPT.x, intersectionPT.y)) {
          return false;
        }
        if (hole.poly.contains(intersectionPT.x, intersectionPT.y)) {
          return false;
        }
      }
      return true;
    }
```
I see two potential issues: 
1. The contains test is repeated for no reason that I can see.
2. I would expect that once a hole for the point is found, the function would return true, and otherwise return false at the end.

So should this function be re-written as follows?
```js
    //Check if a floor is hollowed by a hole
    function checkForHole(intersectionPT, zz) {
      for (let hole of _levels.levelsHoles) {
        const hbottom = hole.range[0];
        const htop = hole.range[1];
        if (zz > htop || zz < hbottom) continue;
        if (hole.poly.contains(intersectionPT.x, intersectionPT.y)) {
          return true;
        }
      }
      return false;
    }
```

**Comment by theripper93** [comment]
while you are correct that there is a repeated check wich is a lefttover of a previous iteration, the returns are correct, if you invert them the holes would break. i can say that the name of the function is knida confusing. but basically if the point is in the poly we want to return false because there is no collision

**Comment by caewok** [comment]
Got it; thanks!


Sent from Yahoo Mail for iPhone


On Friday, July 9, 2021, 14:38, theripper93 ***@***.***> wrote:




Closed #30.
